app/controllers/form_question_controller.rb

```diff
@@ -3,28 +3,79 @@ class FormQuestionController < ApplicationController
   def show
     @form_type = params[:type]
     @list_of_questions = FormQuestion.get_questions_for_form(@form_type)
-  	if (params[:submit]) 
-      if form_completed?
-        flash[:notice] = "#{@form_type} successfully submitted."
-      	redirect_to user_path()
-      else
-      	@error = "Please complete the form before submitting."
+    application = User.find(params[:id]).get_most_recent_application
+    if application
+      @saved_form = application.content
+      if @saved_form.has_key?(@form_type)
+        @form_answer = get_answers_to_prefill_from(@saved_form[@form_type])
+      end
+    end
+  end
+
+  def save_progress
+    @user = User.find(params[:id])
+    @form_content = get_form_content
+    @application = @user.get_most_recent_application
+    if params[:commit] == "Save and Return"
+      update_application(false)
+      flash[:notice] = "#{params[:type]} successfully saved."
+      redirect_to user_path(@user)
+    elsif params[:commit] == 'Submit'
+      submit
+    end
+  end
+
+  def get_answers_to_prefill_from(content)
+    number_of_questions = Form.where(form_name: @form_type)[0].number_of_questions
+    form_answer = Hash.new
+    cur_num_questions = 0
+    content.each do |key, value|
+      if cur_num_questions != number_of_questions && key != "completed"
+        form_answer[cur_num_questions.to_s] = value
+        cur_num_questions += 1
       end
-    elsif (params[:save])
-      save_progress
-      flash[:notice] = "#{@form_type} successfully saved."
-      redirect_to user_path()
     end
+    form_answer
   end
 
-  private
   #filter that check if the form is filled correctly
   def form_completed?
-  	true
+  	!(@form_content[params[:type]].has_value?("") || @form_content[params[:type]].has_value?(nil))
+  end
+  
+  # Form an array of answers using the params
+  def get_answers
```

Comment:
can we just do this? params[:form_answer].values
http://stackoverflow.com/questions/9560335/ruby-hash-to-array-of-values